Wikipedia/Code/NewArticleTabCoordinator.swift

```diff
@@ -135,25 +138,74 @@ final class NewArticleTabCoordinator: Coordinator {
                         relatedArticles: relatedRecords
                     )
                     becauseVM?.onTapArticle = onTapArticleAction
-
                 }
             }
-
+            
+            let dykVM = WMFNewArticleTabDidYouKnowViewModel(
+                fromSourceDefault: self.fromWikipediaDefault,
+                languageCode:  self.dataStore.languageLinkController.appLanguage?.languageCode,
+                dykLocalizedStrings: WMFNewArticleTabDidYouKnowViewModel.LocalizedStrings.init(
+                dyk: self.dyk,
+                fromSource: self.fromLanguageWikipediaTextFor(languageCode: self.dataStore.languageLinkController.appLanguage?.languageCode),
+                ))
+            
             let viewModel = WMFNewArticleTabViewModel(
                 title: CommonStrings.newTab,
-                becauseYouReadViewModel: becauseVM
+                becauseYouReadViewModel: becauseVM,
+                dykViewModel: dykVM
             )
+
             let vc = WMFNewArticleTabViewController(
                 dataStore: self.dataStore,
                 theme: self.theme,
                 viewModel: viewModel
             )
-            
+
+            self.fetchDYK { facts in
+                DispatchQueue.main.async {
+                    dykVM.facts = facts?.map { $0.html }
+                }
+            }
```

Comment:
You should only create the did you know view model if the call succeeds (whether you're fetching from the API or the cache)
If the facts are nil, you won't need the view model.
So you can first fetch the facts, and if it succeeds, you add the view model to the WMFNewArticleTabViewModel